std/encoding/base64.ts

```diff
@@ -1,41 +1,139 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
 /**
- * Converts given data with base64 encoding
- * @param data input to encode
+ * Encodes a given string into RFC4648 base64 representation
+ * @param str
  */
-export function encode(data: string | ArrayBuffer): string {
-  if (typeof data === "string") {
-    return btoa(data);
-  } else {
-    const d = new Uint8Array(data);
-    let dataString = "";
-    for (let i = 0; i < d.length; ++i) {
-      dataString += String.fromCharCode(d[i]);
-    }
+export function encodeString(str: string): string {
+  return btoa(
+    encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function (match, p1) {
+      return String.fromCharCode(parseInt(p1, 16));
+    }),
+  );
+}
 
-    return btoa(dataString);
-  }
+/**
+ * Decodes a given RFC4648 base64 representation into a string
+ * @param b64
+ */
+export function decodeString(b64: string): string {
+  return decodeURIComponent(
+    Array.prototype.map
+      .call(atob(b64), function (c) {
+        return "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2);
+      })
+      .join(""),
+  );
 }
 
+const base64abc = [
```

Comment:
please widen it out to 80 cols manually and use `// deno-fmt-ignore` before the block to disable reformatting... the vertical space is really annoying.

Reply:
Another possibility would be to do it similar to std/encoding/base32:

```ts
const base64abc = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".split("");
```

Reply:
I like both suggestions. @kitsonk 's suggestions has the advantage that it doesn't use another evaluation. So which one is it deno devs?

Reply:
Fixed